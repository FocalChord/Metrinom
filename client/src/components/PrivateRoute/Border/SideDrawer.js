import { Box, Divider, Drawer, List, ListItem, ListItemIcon, ListItemText } from "@material-ui/core";
import { makeStyles } from "@material-ui/core/styles";
import React from "react";
import { Link } from "react-router-dom";

const drawerWidth = 260;

const useStyles = makeStyles((theme) => ({
    drawer: {
        [theme.breakpoints.up("sm")]: {
            width: drawerWidth,
            flexShrink: 0,
        },
    },
    title: {
        flexGrow: 1,
    },
    drawerPaper: {
        width: drawerWidth,
    },
}));

const SideDrawer = ({ routes }) => {
    const classes = useStyles();

    const drawerContent = (
        <div>
            <Box style={{ padding: 10, textAlign: "center" }}>
                <img alt={"Metrinom Logo"} src={MetrinomNoIcon} />
            </Box>
            <Divider />
            <List>
                {routes.map((route, index) => (
                    <ListItem component={Link} button key={index} to={route.path}>
                        <ListItemIcon>{route.icon}</ListItemIcon>
                        <ListItemText primary={route.name} />
                    </ListItem>
                ))}
            </List>
        </div>
    );

    return (
        <nav className={classes.drawer}>
            <Drawer
                classes={{
                    paper: classes.drawerPaper,
                }}
                variant="permanent"
                open
            >
                {drawerContent}
            </Drawer>
        </nav>
    );
};

export default SideDrawer;
