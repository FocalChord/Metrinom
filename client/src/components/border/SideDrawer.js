import { Box, Divider, Drawer, List, ListItem, ListItemIcon, ListItemText, Typography } from "@material-ui/core";

import HistoryIcon from "@material-ui/icons/History";
import { Link } from "react-router-dom";
import MicIcon from "@material-ui/icons/Mic";
import MusicNoteIcon from "@material-ui/icons/MusicNote";
import React from "react";
import { makeStyles } from "@material-ui/core/styles";

const routes = [
    { name: "Home", path: "/", icon: <MicIcon /> },
    { name: "Top Genres", path: "/genres", icon: <MusicNoteIcon /> },
    { name: "Top Artists", path: "/artists", icon: <MusicNoteIcon /> },
    { name: "Top Tracks", path: "/tracks", icon: <MusicNoteIcon /> },
    { name: "Recently Played", path: "/recent", icon: <HistoryIcon /> },
    { name: "Profile", path: "/profile/:user", icon: <HistoryIcon /> },
];

const drawerWidth = 260;

const useStyles = makeStyles((theme) => ({
    drawer: {
        [theme.breakpoints.up("sm")]: {
            width: drawerWidth,
            flexShrink: 0,
        },
    },
    title: {
        flexGrow: 1,
    },
    drawerPaper: {
        width: drawerWidth,
    },
}));

const SideDrawer = () => {
    const classes = useStyles();

    const drawerContent = (
        <div>
            <Box style={{ padding: 10, textAlign: "center" }}>
                <Typography variant="h4" gutterBottom className={classes.title}>
                    Metrinom
                </Typography>
            </Box>
            <Divider />
            <List>
                {routes.map((route, index) => (
                    <ListItem component={Link} button key={index} to={route.path}>
                        <ListItemIcon>{route.icon}</ListItemIcon>
                        <ListItemText primary={route.name} />
                    </ListItem>
                ))}
            </List>
        </div>
    );

    return (
        <nav className={classes.drawer}>
            <Drawer
                classes={{
                    paper: classes.drawerPaper,
                }}
                variant="permanent"
                open
            >
                {drawerContent}
            </Drawer>
        </nav>
    );
};

export default SideDrawer;
