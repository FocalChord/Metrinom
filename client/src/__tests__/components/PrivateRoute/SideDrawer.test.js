import React from "react";
import renderer from "react-test-renderer";
import "regenerator-runtime/runtime.js";
import SideDrawer from "../../../components/PrivateRoute/Border/SideDrawer";

test("<SideDrawer />", () => {
    const tree = renderer.create(<SideDrawer routes={[]} />).toJSON();

    expect(tree).toMatchSnapshot();
});
