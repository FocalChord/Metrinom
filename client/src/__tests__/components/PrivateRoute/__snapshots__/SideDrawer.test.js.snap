// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideDrawer /> 1`] = `
<nav
  className="makeStyles-drawer-1"
>
  <div
    className="MuiDrawer-root MuiDrawer-docked"
  >
    <div
      className="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-3 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
    >
      <div>
        <div
          className="MuiBox-root MuiBox-root-45"
          style={
            Object {
              "padding": 10,
              "textAlign": "center",
            }
          }
        >
          <img
            alt="Metrinom Logo"
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
        <hr
          className="MuiDivider-root"
        />
        <ul
          className="MuiList-root MuiList-padding"
        />
      </div>
    </div>
  </div>
</nav>
`;
